orodha-keycloak/orodha_keycloak/connection.py

```diff
@@ -0,0 +1,129 @@
+
+
+class KeycloakConnection:
+    """
+        Facade class used for connecting to, and interacting with keycloak for the Orodha
+    shopping list app.
+
+    Args:
+        connection_kwargs(dict):
+            server_url(str): The url of the server that our keycloak is hosted at
+            username(str): The username of the Admin server_user.
+            password(str): The password of the Admin server_user.
+            realm_name(str): The name of the keycloak realm that we are attempting to access.
+            client_id(str): The keycloak client_id that we are using for the connection.
+            client_secret_key(str): The secret key of the keycloak client.
+
+    Raises:
+        InvalidConnectionException: If the connection variables given are invalid
+            and do not allow connection.
+    """
+
+    def __init__(self, **connection_kwargs):
+        args = {key: value for key, value in connection_kwargs.items()}
+        # client_connection = create_client_connection(args)
+        admin_connection = orodha_keycloak.connections.admin.create_admin_connection(args)
+        client_connection = orodha_keycloak.connections.client.create_client_connection(args)
+
+        self.admin_connection = admin_connection
+        self.client_connection = client_connection
+
+    def add_user(self, **user_info):
+        """
+        Adds a user to keycloak with a password.
+
+        Args:
+            user_info(kwargs):
+                email(str): The email of the new user.
+                username(str): The username of the new user.
+                firstName(str): The first name of the new user.
+                lastName(str): The last name of the new user.
+                password(str): The password of the new user.
+
+        Returns:
+            new_user: The new user info genereated by the keycloak server.
+
+        """
+        new_user = self.admin_connection.create_user(
+            {
+                "email": user_info.get("email"),
+                "username": user_info.get("username"),
+                "enabled": True,
+                "firstName": user_info.get("firstName"),
+                "lastName": user_info.get("lastName"),
+                "credentials": [
+                    {
+                        "value": user_info.get("password"),
+                        "type": "password",
+                    }
+                ],
+            },
+            exist_ok=False,
+        )
+
+        return new_user
+
+    def delete_user(self, user_id):
+        """
+        Deletes a keycloak user with a given user_id.
+
+        Args:
+            user_id(str): The user id of the user to be deleted.
+
+        Returns:
+            response: The response from the keycloak server with info about the deletion.
+        """
+        response = self.admin_connection.delete_user(user_id=user_id)
+
+        return response
+
+    def check_if_user_exists(self, user_identification, is_token=False):
```

Comment:
@Edison-Stuart If we're returning the user data, perhaps this could be named `get_user_from_access_token`, with the error condition indicating that the user doesn't exist.